src/DynamoPackages/PackageLoader.cs

```diff
@@ -37,6 +37,38 @@ public class PackageLoader : LogSourceBase
         internal event Func<string, IExtension> RequestLoadExtension;
         internal event Action<IExtension> RequestAddExtension;
 
+        private HashSet<string> packagesToIsolate = null;
+        internal HashSet<string> PackagesToIsolate
+        {
+            get
+            {
+                if (packagesToIsolate == null)
+                {
+                    packagesToIsolate = [];
+
+                    string pkgs = DynamoModel.FeatureFlags?.CheckFeatureFlag("IsolatePackages", string.Empty);
+                    if (!string.IsNullOrEmpty(pkgs))
+                    {
+                        foreach (var x in pkgs.Split(","))
+                        {
+                            packagesToIsolate.Add(x);
+                        }
+                    }
+                     
+
+                    pkgs = DynamoModel.FeatureFlags?.CheckFeatureFlag("DoNotIsolatePackages", string.Empty);
```

Comment:
Who specifies which packages should be isolated and which shouldn't? If we do, how do we decide that? Is this meant for only the packages that are owned by us?  If not, how would this be scalable? Am I right in assuming that in the long run, we'll be getting rid of these feature flags to load all packages in isolation by default (except for the ones that already use ALCs, i.e. if there is a way for us to know that ahead of time)?

Reply:
I foresee that some packages might have issues with being loaded in isolation (maybe the interaction between the isolated types vs the ones in the main load context).
So the main reason for adding feature flags is to slowly roll it out for all packages and have the option to disable it for a select few(like the ones that are already isolated on the package side )

As for “how do we decide which ones to isolate”:
I was thinking to take a couple at a time from the list I used to notify package maintainers about the dotnet8 migration.